src/lib/apiTypesV2/Subscription.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include "Subscription.h"
+
+#include <string>
+#include <sstream>
+#include <vector>
+
+#include "common/JsonHelper.h"
+
+using std::string;
+using std::vector;
+
+string Subscription::toJson()
```

Comment:
Please.
What you say is true - most of the function headers only mark where the function begins.
ALL of them should have info on return values, parameters and the implementation.
We haven't had time for that I'm afraid. More complex functions do have a decent function
header, but most functions do not.
All functions have their header though.   It's a start ...